src/lib/component/autocomplete.js

```diff
@@ -0,0 +1,217 @@
+import debounce300 from './debounce300'
+
+export default class Autocomplete {
+  #inputElement
+  #onSearch
+  #onSelect
+  #onPreview
+  #resultsElement
+  #results
+  #highlightedIndex
+  #originalInput
+
+  constructor(inputElement, onSearch, onSelect, onPreview) {
+    this.#inputElement = inputElement
+    this.#onSearch = debounce300(onSearch)
+    this.#onSelect = onSelect
+    this.#onPreview = onPreview
+
+    this.#resultsElement = document.createElement('ul')
+    this.#resultsElement.setAttribute('popover', 'auto')
+    this.#resultsElement.classList.add('textae-editor__dialog__autocomplete')
+    inputElement.parentElement.appendChild(this.#resultsElement)
+
+    this.#highlightedIndex = -1 // Initialize keydown pointer.
+    this.#results = []
+
+    this.#inputElement.addEventListener('input', (event) =>
+      this.#handleInput(event.target.value)
+    )
+    this.#inputElement.addEventListener('keydown', (event) =>
+      this.#handleKeyDown(event)
+    )
```

Comment:
よく考えるとdebounceしたいのは、検索ではなく入力の連打でした。
ですのでここにdebounceを入れるのが良さそうです。

```suggestion
    var handleInput = debonce300(this.#handleInput)
    this.#inputElement.addEventListener('input', (event) => handleInput(event.target.value))
    var handleKeydown = debounce300(this.#handleKeydown)
    this.#inputElement.addEventListener('keydown', handleKeydown)
```

上のモデルクラスを導入したら

```suggestion
    var handleInput = debonce300((term) => model.term = term)
    this.#inputElement.addEventListener('input', (event) => handleInput(event.target.value))
    var handleKeydown = debounce300(this.#handleKeydown)
    this.#inputElement.addEventListener('keydown', handleKeydown)
```

で、行けそうです。
Keydownの方はキーの種類の意味を解釈するロジックが複雑なので、メソッドのままにした方が良さそうです。

Reply:
inputにdebounceを適応しました。
https://github.com/pubannotation/textae/pull/142/commits/6070588a377252a16de396dc50fb0c24fca9b2e9

キー操作にdebounceを入れると、キー操作自体の反応が遅くなってしまうのでキー操作は変更しませんでした。
解釈が間違っていたらすみません。

Reply:
御指摘の通り、キー操作のほうはdebounce不要でした。